src/balancerd/src/lib.rs

```diff
@@ -547,6 +561,41 @@ impl mz_server_core::Server for PgwireBalancer {
     }
 }
 
+async fn cancel_request(conn_id: u32, secret_key: u32, cancellation_resolver: &str) {
```

Comment:
Can you add a comment about how cancellation works and why it's safe to broadcast the cancellation request to multiple environments?